File: c++/int_to_array.cpp
// https://programmers.co.kr/learn/courses/30/lessons/12932

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(long long n)
{
    vector<int> answer;
    string tmp = to_string(n);
    reverse(tmp.begin(), tmp.end());
    for (char c : tmp)
    {
        answer.push_back(c - '0');
    }

    return answer;
}

// vector<int> solution(long long n)
// {
//     vector<int> answer;
//     while (n)
//     {
//         answer.push_back(n % 10);
//         n /= 10;
//     }
//     return answer;
// }

File: java/int_to_array.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12932

import java.util.ArrayList;

class Solution {
    public int[] solution(long n) {
        ArrayList<Long> arrList = new ArrayList<>();
        while (n != 0) {
            arrList.add(n % 10);
            n /= 10;
        }
        return arrList.stream().mapToInt(Long::intValue).toArray();
    }
}

// class Solution {
//     public int[] solution(long n) {
//         String a = "" + n;
//         int[] answer = new int[a.length()];
//         int cnt=0;

//         while(n>0) {
//             answer[cnt]=(int)(n%10);
//             n/=10;
//             System.out.println(n);
//             cnt++;
//         }
//         return answer;
//     }
// }

// class Solution {
//     public int[] solution(long n) {
//         String s = String.valueOf(n);
//         StringBuilder sb = new StringBuilder(s);
//         sb = sb.reverse();
//         String[] ss = sb.toString().split("");

//         int[] answer = new int[ss.length];
//         for (int i=0; i<ss.length; i++) {
//             answer[i] = Integer.parseInt(ss[i]);
//         }
//         return answer;
//     }
// }